tests/BenchmarkDotNet.IntegrationTests.ManualRunning/ExpectedBenchmarkResultsTests.cs

```diff
@@ -0,0 +1,220 @@
+            );
+
+            var cpuGhz = RuntimeInformation.GetCpuInfo().MaxFrequency.Value.ToGHz();
+
+            foreach (var report in summary.Reports)
+            {
+                Assert.True(cpuGhz * report.ResultStatistics.Mean < 1, $"Actual time was greater than 1 clock cycle.");
+
+                var overheadTime = report.AllMeasurements
+                    .Where(m => m.IsOverhead() && m.IterationStage == Engines.IterationStage.Actual)
+                    .Select(m => m.GetAverageTime().Nanoseconds)
+                    .Average();
+
+                var workloadTime = report.AllMeasurements
+                    .Where(m => m.IsWorkload() && m.IterationStage == Engines.IterationStage.Actual)
+                    .Select(m => m.GetAverageTime().Nanoseconds)
+                    .Average();
+
+                // Allow for 1 cpu cycle variance
+                Assert.True(overheadTime * cpuGhz < workloadTime * cpuGhz + 1, "Overhead took more time than workload.");
+
+                Assert.True((report.GcStats.GetBytesAllocatedPerOperation(report.BenchmarkCase) ?? 0L) == 0L, "Memory allocations measured above 0.");
+            }
+        }
+
+        [Fact]
+        public void LargeStructBenchmarksReportsNonZeroTimeAndZeroAllocated_InProcess()
+        {
+            AssertLargeStructResults(ManualConfig.CreateEmpty()
+                .AddJob(Job.Default
+                    .WithToolchain(InProcessEmitToolchain.Instance)
+                ));
+        }
+
+        [TheoryNetCoreOnly("To not repeat tests in both full Framework and Core")]
+        [InlineData(RuntimeMoniker.Net70)]
+        [InlineData(RuntimeMoniker.Mono70)]
+        public void LargeStructBenchmarksReportsNonZeroTimeAndZeroAllocated_Core(RuntimeMoniker runtimeMoniker)
+        {
+            AssertLargeStructResults(ManualConfig.CreateEmpty()
+                .AddJob(Job.Default
+                    .WithRuntime(runtimeMoniker.GetRuntime())
+                ));
+        }
+
+        [TheoryFullFrameworkOnly("Can only run full Framework and Mono tests from Framework host")]
+        [InlineData(RuntimeMoniker.Net462)]
+        [InlineData(RuntimeMoniker.Mono)]
+        public void LargeStructBenchmarksReportsNonZeroTimeAndZeroAllocated_Framework(RuntimeMoniker runtimeMoniker)
+        {
+            AssertLargeStructResults(ManualConfig.CreateEmpty()
+                .AddJob(Job.Default
+                    .WithRuntime(runtimeMoniker.GetRuntime())
+                ));
+        }
+
+        private void AssertLargeStructResults(IConfig config)
+        {
+            var summary = CanExecute<LargeStruct>(config
+                .WithSummaryStyle(SummaryStyle.Default.WithTimeUnit(Perfolizer.Horology.TimeUnit.Nanosecond))
+                .AddDiagnoser(new MemoryDiagnoser(new MemoryDiagnoserConfig(false)))
+            );
+
+            var cpuGhz = RuntimeInformation.GetCpuInfo().MaxFrequency.Value.ToGHz();
```

Comment:
MaxFrequency is not available in some environments. See how we workaround this problem in `ZeroMeasurementAnalyser` with `FallbackCpuResolutionValue`